openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -191,110 +194,393 @@ def test_get_program_for_certificates_id_not_exist(self):
+        self.assertEqual(
+            meter.progress,
+            [{
+                'id': program['id'],
+                'completed': 1,
+                'in_progress': 1,
+                'not_started': 0,
+            }]
+        )
 
-        self.assertEqual(expected, actual)
+        # Certificate still invalid, still one course code to complete.
+        second_certificate.mode = GeneratedCertificate.MODES.verified
+        second_certificate.status = CertificateStatuses.audit_passing
+        second_certificate.save()  # pylint: disable=no-member
+        self.assertEqual(
+            meter.progress,
+            [{
+                'id': program['id'],
+                'completed': 1,
+                'in_progress': 1,
+                'not_started': 0,
+            }]
+        )
+
+        # Second valid certificate obtained, all course codes complete.
+        second_certificate.status = CertificateStatuses.downloadable
+        second_certificate.save()  # pylint: disable=no-member
+        self.assertEqual(
+            meter.progress,
+            [{
+                'id': program['id'],
+                'completed': 2,
+                'in_progress': 0,
+                'not_started': 0,
+            }]
+        )
 
     @httpretty.activate
-    def test_get_display_category_success(self):
-        self.create_programs_config()
-        self.mock_programs_api()
-        actual_programs = get_programs(self.user)
-        for program in actual_programs:
-            expected = 'XSeries'
-            self.assertEqual(expected, get_display_category(program))
+    def test_nonstandard_run_mode_completion(self):
+        """
+        A valid run mode isn't necessarily verified. Verify that the program can
+        still be completed when this is the case.
+        """
+        course_id = 'org/course/run'
+        data = [
+            factories.Program(
+                organizations=[factories.Organization()],
+                course_codes=[
+                    factories.CourseCode(run_modes=[
+                        factories.RunMode(
+                            course_key=course_id,
+                            mode_slug=GeneratedCertificate.MODES.honor
+                        ),
+                    ]),
+                ]
+            ),
+        ]
+        self._mock_programs_api(data)
 
-    def test_get_display_category_none(self):
-        self.assertEqual('', get_display_category(None))
-        self.assertEqual('', get_display_category({"id": "test"}))
+        enrollments = self._create_enrollments(course_id)
+        meter = ProgramProgressMeter(self.user, enrollments)
+
+        self._grant_certificate(course_id, mode=GeneratedCertificate.MODES.honor)
+
+        program = data[0]
+        self.assertEqual(
+            meter.progress,
+            [{
+                'id': program['id'],
+                'completed': 1,
+                'in_progress': 0,
+                'not_started': 0,
+            }]
+        )
```

Comment:
this assert is repeated a lot and probably deserves a helper method for shortand, e.g. `self.assert_progress(meter, program['id'], 1, 0, 0)`